go/vt/tabletserver/sqlquery.go

```diff
@@ -175,12 +190,12 @@ func (sq *SqlQuery) disallowQueries() {
 	sq.mu.Unlock()
 	// Terminate all streaming queries
 	sq.qe.streamQList.TerminateAll()
-	// Don't hold lock while waiting.
+	// Wait for outstanding requests to finish.
 	sq.requests.Wait()
 
-	// SHUTTING_QUERIES -> NOT_SERVING
-	sq.mu.Lock()
 	defer func() {
```

Comment:
Yeah. This area is nebulous. If qe.Close failed, we don't know if we're in a recoverable state.
Maybe I should crash the server on failure.